File: gateway/dakota_gateway/crypto.py
from __future__ import annotations

import hashlib
import hmac


def sha256_hex(data: bytes) -> str:
    return hashlib.sha256(data).hexdigest()


def hmac_sha256_hex(key: bytes, data: bytes) -> str:
    return hmac.new(key, data, hashlib.sha256).hexdigest()

File: gateway/internal/audit/crypto.go
package audit

import (
	"crypto/hmac"
	"crypto/sha256"
	"encoding/hex"
)

func sha256Hex(b []byte) string {
	sum := sha256.Sum256(b)
	return hex.EncodeToString(sum[:])
}

func hmacSHA256Hex(key []byte, b []byte) string {
	m := hmac.New(sha256.New, key)
	_, _ = m.Write(b)
	return hex.EncodeToString(m.Sum(nil))
}

